### ISSUE TITLE
Double checking accuracy of predictions -- statistics window false positive counts

### ISSUE THREAD

**Issue by maxieds** [gtDMMB/RNAStructViz#32]
@ceheitsch has suggested that we double check the *false positive counts* in the stats window. The two screenshots below correspond to the relevant windows for *C. elegans*:

![Screenshot at 2019-03-10 23-42-11](https://user-images.githubusercontent.com/22165688/54099194-6460e700-438e-11e9-82fd-8e5d5ba184b7.png)
![Screenshot at 2019-03-10 23-42-24](https://user-images.githubusercontent.com/22165688/54099195-6460e700-438e-11e9-92e1-5a1ad0cbb987.png)

**Comment by annakirkpatrick** [comment]
@maxieds What exactly do you need help with here? (Or, alternatively, what don't you understand?)

As I have explained, I have many other responsibilities and am currently dealing with a chronic pain flare. I do not have the bandwidth (or, at the moment, physical capacity) to write some scripts to parse the structures and recalculate these values.

**Comment by maxieds** [comment]
@annakirkpatrick 
This is one of the checking tasks that @ceheitsch flagged for me to enter two Wednesdays ago. It's not urgent. What am I looking to check here?

**Comment by annakirkpatrick** [comment]
@maxieds I think what you're missing here are just some definitions. Let me know if I'm wrong.

You should think of all of this as set theory. Base pairs are the elements of our sets.

True positives = base pairs in reference structure and predicted structure
False positives = base pairs in predicted structure but not in reference structure
False negatives = base pairs in reference structure but not in predicted structure

I think Christine wanted you to check that those values are indeed being calculated according to the above definitions. You probably want to select a simpler (shorter) sequence if you want to check these values by hand. You could also just do a little scripting to extract these sets of base pairs from the various CT files, take the appropriate intersections and set differences yourself, and compare to the results that RNAStructViz returns. And if anything doesn't match, of course you then need to look back at the code that generates these bar graphs.

**Comment by maxieds** [comment]
@ceheitsch This issue should now be resolved. See the screenshot below for the small mock example I created to test these calculations:

![Screenshot at 2019-07-20 02-45-55](https://user-images.githubusercontent.com/22165688/61575359-9d296f00-aa98-11e9-980c-f08f0a3a45fd.png)

These correspond to the two following DB (DOT) files:
```
> Another C. Elegans sample
UAAAGUUUUCUUUCAGGGAAUUAAAAUUUGAUCAUGGUUU
().((((..(.....)..)))).......(((((.)))))

> Start of C. Elegans
UAAAGUUUUCUUUCAGGGAAUUAAAAUUUGAUCAUGGUUU
().....................(((((.......)))))
```
I believe that the discrepancies Christine noticed is not an actual computational error (as in the formulas are coded incorrectly in the source), but rather a misleading initial count for TP/FP/FN corresponding to the reference structure. Observe, in the *True Positives* graph above, even though there is only one TP to count in the predicted structures, there is still the **total possible count** of 11 in black. This is not the best way to do things, but for the historical users, let's leave it as is. I added a clarifying legend image on the RHS of the StatsWindow to explain (as @annakirkpatrick defined above) to users what these bars and counts actually reflect. 

Now closing this long standing issue since it has been tested and deemed resolved sanely.
